torch/utils/data/datapipes/_hook_iterator.py

```diff
@@ -138,7 +143,7 @@ def __next__(self):
             # TODO: Add try-except to in-place reduce traceback from the Exception
             # See: https://github.com/pytorch/data/issues/284
             if self._profiler_enabled:
-                with profiler_record_fn_context():
+                with profiler_record_fn_context(self.source_dp):
```

Comment:
Is `self.source_dp` the `DataPipe` that generates this iterator or the one prior to this `DataPipe`?

Reply:
Okay, I think it's the `DataPipe` that generates this iterator. Can we change the name from `source_dp` to `datapipe`?